### PR TITLE
feat: no-buffer split

### PR DESCRIPTION
**Pull request by fortuna**
The ReadFrom call no longer needs to allocate a buffer now. The buffer is provided by the inner ReaderFrom.
We should prefer ReadFrom when possible to avoid unnecessary buffers, but it needs to be properly implemented. This is an example of how we can accomplish that.

### CODE REVIEW

**Review by fortuna — COMMENTED** [review]
Thoughts?

File: transport/split/reader_test.go [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestSplitReader(t *testing.T) {
+	splitReader := &splitReader{bytes.NewReader([]byte("Request")), 3}
+	buf := make([]byte, 10)
+	n, err := splitReader.Read(buf)
+	require.NoError(t, err)
+	require.Equal(t, int(3), n)
```
Comment (jyyi1) [inline transport/split/reader_test.go:29]: *minor* do we actually need the cast here?
```suggestion
	require.Equal(t, 3, n)
```

File: transport/split/reader_test.go [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+	require.Equal(t, []byte("Req"), buf[:n])
+
+	n, err = splitReader.Read(buf)
+	require.NoError(t, err)
+	require.Equal(t, int(4), n)
```
Comment (jyyi1) [inline transport/split/reader_test.go:34]: *minor* same as above.
```suggestion
	require.Equal(t, 4, n)
```

File: transport/split/writer.go [hunk 3]
```diff
@@ -15,42 +15,42 @@
...
-			return written, nil
-		}
-		if err != nil {
-			return written, err
-		}
-	}
-	n, err := io.Copy(w.writer, source)
-	written += n
+func (w *splitWriterReaderFrom) ReadFrom(source io.Reader) (int64, error) {
+	written, err := w.ReaderFrom.ReadFrom(&splitReader{source, w.prefixBytes})
```
Comment (jyyi1) [inline transport/split/writer.go:48]: An alternative way is to reuse `io`'s standard readers. I'm ok with both, you can choose whichever one you prefer.

```
limit := io.LimitReader(source, w.prefixBytes)
r := io.MultiReader(limit, source)
written, err := w.ReaderFrom.ReadFrom(r)
```

**Review by jyyi1 — APPROVED** [review]
Code is good, with some minor suggestions.

File: transport/split/writer.go (hunk 3 above)
Comment (fortuna) [reply transport/split/writer.go]: Sounds good. Less code to maintain.

**Pull request merged**